mozilla/addons-server
# Sync storage with make up flow

Relates to: mozilla/addons#15066
Child of: https://github.com/mozilla/addons-server/pull/22781

### Description

Sync the `storage` director with make initialize command. 1) when you dump/load a db backup it will store/load the storage directory as well (correctly) 2) when seeding the db it will sync the storage directory as well 3) when syncing the storage directory it will clear the directory first to ensure pristine backups 4) remove the storage volume and just mount the repo and map the files correctly to static or storage directories.

### Context

Currently `data_seed` is semi broken as it does not sync the storage directory when creating a snapshot.

### Testing

run a clean init

> NOTE: this will remove your existing DB so create a backup if you want to restore after

```bash
make up INIT_CLEAN=True
```

Expect the storage directory to be cleared and populated with the data from addons created during the seed.

run a backup and restore.

1. have a running environment
2. upload an addon
3. create a backup `make data_dump`
4. upload another addon
5. run a restore `make data_load`
6. expect the files associated with your 1st addon to be restored but not the 2nd.

### Checklist
